packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx

```diff
@@ -88,33 +118,74 @@ const PdfViewer: SFC<Props> = ({
     };
   }, [loadedFile, page]);
 
-  const [viewport, canvasInfo] = useMemo(() => {
-    const viewport = loadedPage?.getViewport({ scale });
-    const canvasInfo = viewport ? getCanvasInfo(viewport) : undefined;
-    return [viewport, canvasInfo];
-  }, [loadedPage, scale]);
+  const currentPage = useMemo(() => {
+    const isPageValid = !!loadedPage && loadedPage.pageNumber === page;
+    if (isPageValid) {
+      const viewport = loadedPage?.getViewport({ scale });
+      const canvasInfo = viewport ? getCanvasInfo(viewport) : undefined;
+      return { loadedPage, viewport, canvasInfo };
```

Comment:
This will make the code confusing, as in the following section: there's the main `loadedPage`, and then there's the one from within `currentPage`.

Actually, there isn't a reason to store `loadedPage` within `currentPage`, since it's already available in this context. Is there a reason for going away from the previous code, which added `viewport` and `canvasInfo` at the main closure level?

If there is a reason for that, then I suggest we instead name this `loadedPageInfo` and set it to `{ viewport, canvasInfo }`.